### PR TITLE
[APPVEYOR] Create appveyor.yml

### PR DESCRIPTION
**Pull request by SergeGautherie**
To leverage this PR, a `ReactOS` admin should register `sysreg2` on `AppVeyor`.

### CODE REVIEW

**Comment by learn-more** [comment]
What is the purpose of this PR?
What do we gain by running this in appveyor?

**Comment by SergeGautherie** [comment]
@learn-more, I am puzzled by your questions,
as I assume adding "Continuous Integration" speaks for itself, especially as we already have it on `ReactOS` itself.

**Comment by sanchaez** [comment]
We could use TravisCI or other things, why AppVeyor?

**Comment by SergeGautherie** [comment]
> We could use TravisCI or other things, why AppVeyor?

`AppVeyor` is already used for `ReactOS` itself and is the service I use.
Anyone else can add support for other CI service(s) too.

`Sysreg2` project admin may use whichever service(s) is preferred, or none even...

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+image: Ubuntu
+
+version: sysreg2.appveyor.{build}
+skip_branch_with_pr: true
+clone_depth: 5
+clone_folder: /sysreg2_workdir
```
Comment (learn-more) [inline appveyor.yml:6]: This is a weird path for windows?

Comment (SergeGautherie) [reply appveyor.yml:6]: Ubuntu is Linux, not Windows...

Comment (learn-more) [reply appveyor.yml:6]: Not sure what I was smoking yesterday.

**Comment by SergeGautherie** [comment]
@HeisSpiter, why did you revert this commit in 24b7ae6ca2db3b1cf4d23677ad6fac6a915cb015?

**Comment by HeisSpiter** [comment]
Now that travis CI is configured, there's no point of keeping AppVeyor which was never configured with the repository.
As you wrote:
> Sysreg2 project admin may use whichever service(s) is preferred

Job done ;-)

**Comment by SergeGautherie** [comment]
> Job done ;-)

Well, you and I obviously do not have the same point of view:
while I did mean adding Travis CI, and even enabling it, would be fine,
I never meant that should trigger unsupporting community people, like me, who do use AppVeyor :-(

In other words, you did explicitly break my builds (as in making them more complicated to set up), removed that "template", and are implicitly trying to "force" me to use Travis CI:
that is exactly the kind of decisions I call "developer-centric, community-excluding" 👎

Unless you could tell me what harm that file would be causing, that I totally missed?

**Comment by learn-more** [comment]
That is what happens if you give vague answers when a developer asks what the point of a PR is.

You said:
'as I assume adding "Continuous Integration" speaks for itself, especially as we already have it on ReactOS itself.'

instead of:
'I want to use CI on my own branch, and this change makes it easier.'

Consider that you are talking to humans, not to machines, so we do not ask for something that we can google (what is CI), but we ask for the reasoning behind it.

**Comment by HeisSpiter** [comment]
No overreaction, please. Feel free to use whatever you want in your local branch. I see you already have a fully configured AppVeyor in your branch. That's perfect, that will even be easier for you to maintain as there will be no conflicts given the file only exists in your branch.

**Comment by SergeGautherie** [comment]
> if you give vague answers

I wrote "I am puzzled by your questions",
so I believe my answers were no more vague than asked questions seemed to be to me...

Later, I even wrote "AppVeyor [...] is the service I use",
which seems rather explicit to me!

I too am a human, not a machine: as I noted, I did not understand/expect such questions, as the answers seemed so obvious to me :-/

Eventually, you committed this PR, so I assumed the reasoning had become clear enough.

---

> local branch

There were no conflicts so far and I did not expect "any" in the future: at least, no more than on 'reactos' repository.
And, no, endlessly maintaining a local commit does not seem easier to me nor more useful to the community: quite the opposite.
As in: could you un-revert this PR?

My complain would be "I" was not even asked/warned before reverting this PR.
Especially, "could you tell me what harm that file would be causing", even if afterward now.

**Pull request merged**